open-liberty/content.md

```diff
@@ -0,0 +1,148 @@
+	```
+
+	This can then be built and run as follows:
+
+	```console
+	$ docker build -t app .
+	$ docker run -d -p 80:9080 -p 443:9443 app
+	```
+
+4.	You can mount a data volume container that contains the application and the server configuration on to the image. This has the benefit that it has no dependency on files from the host but still allows the application container to be easily re-mounted on a newer version of the application server image. This example assumes that you have copied the `/tmp/DefaultServletEngine` directory in to the same directory as the Dockerfile.
+
+	Build and run the data volume container:
+
+	```dockerfile
+	FROM %%IMAGE%%:webProfile7
+	ADD DefaultServletEngine /config
+	```
+
+	```console
+	$ docker build -t app-image .
+	$ docker run -d -v /config \
+	    --name app app-image true
+	```
+
+	Run the Open Liberty image with the volumes from the data volume container mounted:
+
+	```console
+	$ docker run -d -p 80:9080 \
+	  --volumes-from app %%IMAGE%%:webProfile7
+	```
+
+# Providing your own keystore/truststore
+
+When an `open-liberty` image starts, it can generate a Liberty server XML snippet in `/config/configDropins/defaults/keystore.xml` that specifies a `keyStore` stanza with a generated password. This causes Open Liberty to generate a default keystore and truststore with a self-signed certificate when it starts. The `javaee7` image does this automatically, but other images can request this by setting:
+
+```console
+ENV KEYSTORE_REQUIRED "true"
+```
+
+When providing your own keystore/truststore, this default behavior can be disabled by adding:
+
+```console
+ENV KEYSTORE_REQUIRED "false"
+```
+
+It is good practice to place the keystore customization in `/config/configDropins/defaults/keystore.xml` even when not generated since this makes it easier to find and makes moving to the websphere-liberty docker image simpler.
+
+# Using IBM JRE Class data sharing
+
+The IBM JRE provides a feature [Class data sharing](http://www-01.ibm.com/support/knowledgecenter/SSYKE2_8.0.0/com.ibm.java.lnx.80.doc/diag/understanding/shared_classes.html) which offers transparent and dynamic sharing of data between multiple Java Virtual Machines running on the same host by using shared memory backed by a file. When running the Liberty Docker image, it looks for the file at `/opt/ol/wlp//output/.classCache`. To benefit from Class data sharing, this location needs to be shared between containers either through the host or a data volume container.
+
+Taking the application image from example 3 above, containers can share the host file location (containing the shared cache) `/tmp/open-liberty/classCache` as follows:
+
+```console
+docker run -d -p 80:9080 -p 443:9443 \
+    -v /tmp/open-liberty/classCache:/opt/ol/wlp//output/.classCache app
+```
+
+Or, create a named data volume container that exposes a volume at the location of the shared file:
+
+```console
+docker run -e LICENSE=accept -v /opt/ol/wlp//output/.classCache \
```

Comment:
Does Open Liberty still require license acceptance?  (maybe just another bad copy-paste?)

Reply:
You are right, total copy and paste error.